### DISCUSSION TITLE
WhatsApp Web via API Channels

### DISCUSSION THREAD

**Discussion by antoniomdk** [chatwoot/chatwoot#6551] [Show and tell]
Hey guys, I have put together a small repo with a node service that adds support for Whatsapp Web via API channels.

https://github.com/antoniomdk/whatsapp-chatwoot

I'm working on improving the docs and adding fault tolerance so there are no missing messages if the WhatsApp worker is down.

The service also has `/status` and `/qrcode` endpoints that I call from a custom dashboard app to monitor the session.

The whatsApp-web.js library and my code itself are not production-ready and subject to change, so use it at your own risk, but feedback is much appreciated.

Cheers,

**Comment by joeenet** [comment]
Did you use node js?

**Reply by antoniomdk** [reply]
Yes, it is a node/typescript small server with whatsapp-web.js library.

**Comment by VSinghhh** [comment]
@antoniomdk A little curious: what was the reason you chose to implement WhatsApp Web as an API channel rather than use the build in functionality in chatwoot?

**Reply by antoniomdk** [reply]
Because Chatwoot WhatsApp integration is based on WhatsApp Cloud API, which requires creating a FB app, verifying it, and a couple of other steps. But most importantly, the phone number you use for WhatsApp Business API can't be used on the WhatsApp mobile app, which is a huge problem for my use case.

**Reply by VSinghhh** [reply]
Got it! Thank you for the clarification!

**Comment by ubeyou** [comment]
Hi, thanks for the amazing fork! i curious if it will pull existing app history into the chatwoot?
